src/sage/schemes/curves/affine_curve.py

```diff
@@ -314,52 +314,132 @@ def _repr_type(self):
 
     def divisor_of_function(self, r):
         """
-        Return the divisor of a function on a curve.
-
-        INPUT:
-
-        - ``r`` -- a rational function on X
-
-        OUTPUT:
-
-        - ``list`` -- the divisor of r represented as a list of coefficients
-          and points. (TODO: This will change to a more structural output in
-          the future.)
-
-        EXAMPLES::
-
-            sage: F = GF(5)
-            sage: P2 = AffineSpace(2, F, names='xy')
-            sage: R = P2.coordinate_ring()
-            sage: x, y = R.gens()
-            sage: f = y^2 - x^9 - x
-            sage: C = Curve(f)
-            sage: K = FractionField(R)
-            sage: r = 1/x
-            sage: C.divisor_of_function(r)      # not implemented (broken)
-                  [[-1, (0, 0, 1)]]
-            sage: r = 1/x^3
-            sage: C.divisor_of_function(r)      # not implemented (broken)
-                  [[-3, (0, 0, 1)]]
-        """
-        F = self.base_ring()
-        f = self.defining_polynomial()
-        pts = self.places_on_curve()
-        R = f.parent()
-        x, y = R.gens()
-        R0 = PolynomialRing(F, 3, names=[str(x), str(y), "t"])
-        vars0 = R0.gens()
-        t = vars0[2]
-        divf = []
-        for pt0 in pts:
-            if pt0[2] != F(0):
-                lcs = self.local_coordinates(pt0, 5)
-                yt = lcs[1]
-                xt = lcs[0]
-                ldg = degree_lowest_rational_function(r(xt, yt), t)
-                if ldg != 0:
-                    divf.append([ldg, pt0])
-        return divf
+            Return the divisor of a function on an affine curve.
+    
+            INPUT: r is a rational function on X
+    
+            OUTPUT:
+    
+    
+            -  ``list`` - The divisor of r represented as a list of
+                   coefficients and points. (TODO: This will change to a more
+                   structural output in the future, so
+                   intersection_number should disappear in the future.)
```

Comment:
```suggestion
            -  ``list`` - The divisor of r represented as a list of
                   coefficients and points.
                   
           .. TODO::
           
               This will change to a more structural output in the future, so ``intersection_number`` should disappear in the future.
```

Please wrap this line for me, as GitHub review interface is not very good. `.. TODO::` is a [Sphinx directive](https://www.sphinx-doc.org/en/master/usage/extensions/todo.html) for the documentation :)